subscription.go

```diff
@@ -0,0 +1,219 @@
+package opcua
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/gopcua/opcua/ua"
+)
+
+const (
+	DefaultSubscriptionMaxNotificationsPerPublish = 10000
+	DefaultSubscriptionLifetimeCount              = 10000
+	DefaultSubscriptionMaxKeepAliveCount          = 3000
+	DefaultSubscriptionInterval                   = 100 * time.Millisecond
```

Comment:
These defaults seem OK, but this turns into a 5-minute keepalive interval.  Our default SessionTimeout is 1 minute so our session ends up timing out before the server even gets a chance to send the keepalive.  I recommend increasing the default SessiontTimeout.  There's no firm recommendation in the spec or test cases that I could find, but 20 minutes seems to be the most common with other clients (shortest I saw was 5 minutes, longest was 1 hour).

Reply:
OK. Changed to 20 minutes.

Reply:
For some mysterious reason, there's no reply box under [this comment](https://github.com/gopcua/opcua/pull/207#discussion_r290119821), so I'll reply here.

Do you think we should assume that any error we didn't explicitly ignore is irrecoverable? I tend to agree, but I'm not very sure.